dev_docs/contributing/kibana_http_api_guidelines.mdx

```diff
@@ -1,820 +1,440 @@
+# -> 204 No content
 ```
 
-[Provider configuration example](https://github.com/elastic/terraform-provider-elasticstack/blob/main/examples/provider/provider.tf)
+**Stick to simple methods**
 
-OpenAPI specifications make generating consistent Kibana and Fleet clients possible:
+`GET`, `POST`, `PUT`, `PATCH`, `DELETE` should cover almost all cases. If you are considering `HEAD` or some other method make sure you have a good justification for your HTTP API.
 
-```bash
-# From Makefile - generates standardized clients
-internal/clients/fleet/:
-	oapi-codegen -package fleet fleet-openapi.json
-```
+### Path parameters
 
-[Provider makefile](https://github.com/elastic/terraform-provider-elasticstack/blob/main/Makefile)
+**Should use snake case**
 
-Standardization means authentication, error handling, and request patterns become reusable across all resources.
+`/api/my_api/{my_id}` ✅
 
-### Return as much as you can, and handle defaults carefully
+`/api/my_api/{myId}` ❌
 
-Terraform needs complete information to track resource state effectively. If your API doesn't return enough details, Terraform can't properly detect changes or manage drift.
+**BEWARE: path parameters should not expect values of unknown length**
 
-**The challenge with defaults**
+Accepting very long strings in path parameters can cause issues for HTTP servers that limit the byte length of certain parts of requests.
 
-Many Kibana APIs have extensive configuration options. Requiring users to specify every single value would create verbose, unwieldy configurations. We need to keep in mind that changing a default may cause Terraform to get out-of-sync and so view changes to defaults as a breaking change until we can update our provider.
+Use a body parameter for long strings instead and consider using validation to limit a path param's length.
 
-### Be predictable
+### Query parameters
 
-Terraform relies on the same input resulting in the same output.  Avoid using fields like "last modified time"—Terraform can’t compare those meaningfully. This can be tricky with sensitive fields or when your API relies on randomness.
+**Should use snake case**
 
-Constantly changing fields in API responses create a difficult situation for Terraform implementations because they cause constant drift.
+`/api/my_api?snake_case=true` ✅
 
-Timestamps and other time-based fields aren’t predictable:
+`/api/my_api?snakeCase=false` ❌
 
-```
-settings {
-  setting {
-    name = "index.creation_date"      # Timestamp varies
-    value = "1643651976221"           # Unix timestamp changes
-  }
-}
-```
+**BEWARE: query parameters should not expect values of unknown length**
 
-The [Alerting Rules API](https://www.elastic.co/docs/api/doc/kibana/operation/operation-post-alerting-rule-id) is another example, where execution status data changes on every read:
+Accepting very long strings in query parameters can cause issues for HTTP servers that limit the byte length of certain parts of requests.
```

Comment:
Duplicate(ish) of long string warning for path params. Maybe we can combine path params and query params sections, since it's pretty much identical guidence for both.